pkg/datapath/linux/node.go

```diff
@@ -1356,6 +1367,17 @@ func (n *linuxNodeHandler) NodeConfigurationChanged(newConfig datapath.LocalNode
 				return fmt.Errorf("cannot find link by name %s for neigh discovery: %w",
 					ifaceName, err)
 			}
+
+			// store the neighDiscoveryLink used so that we can remove the ARP
+			// PERM entries created by this configuration run in case it's
+			// necessary
```

Comment:
```suggestion
			// Store neighDiscoveryLink so that we can remove the ARP
			// PERM entries when cilium-agent starts with neigh discovery
			// disabled next time.
```